parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetOutputFormat.java

```diff
@@ -317,6 +319,16 @@ private static int getMaxPaddingSize(Configuration conf) {
     return conf.getInt(MAX_PADDING_BYTES, DEFAULT_MAX_PADDING_SIZE);
   }
 
+  public static ValuesWriterFactory getValuesWriterFactory(Configuration conf) {
+    ValuesWriterFactory factory = ParquetProperties.DEFAULT_VALUES_WRITER_FACTORY;
+
+    if (factory instanceof Configurable) {
+      Configurable configurableFactory = (Configurable) factory;
+      configurableFactory.setConf(conf);
+    }
```

Comment:
Hmm I don't think creating a `ParquetProperties.getValuesWriterFactory(Configuration conf)` is possible cause ParquetProperties is in parquet-column (which doesn't depend on hadoop so we don't have Config there) so we'll have to do this in ParquetOutputFormat. 
Configuration happens by creating a ValuesWriterFactory that implements Configurable:
```
public class MyConfigurableValuesWriterFactory implements ValuesWriterFactory, Configurable {
...
}
```

Now when we create a new ValuesWriterFactory in `getValuesWriterFactory()` via `getRecordWriter(...)` we pass the config object to the factory there.